### DISCUSSION TITLE
How to promote contributing to the configuration and incentivize the use of the website?

### DISCUSSION THREAD

**Discussion by jakopako** [jakopako/croncert-config#254] [General]
This discussion is meant to collect a bunch of thoughts concerning the question in the title and on top of that figure out ways of assessing whether the entire project even makes sense in the sense of 'do people even need such a website?'

Actually, the most important question is probably whether people even need such a website. Or are there enough other websites that provide the information that people need? Does this website/project provide extra value that no other website provides? What are people's preferred ways of finding good concerts? I anyone even interested in finding new venues? Or does everyone rather go to known venues or follow suggestions received from friends etc.?

Let's say there is a need for such a website, then how do we find contributors? The group of people most likely to contribute is probably somewhere in the intersection between people that like music (and are especially curious to find new music / venues) and people that have sufficient IT skills to add events to the concert database by extending the configuration file in this repository. How does this group of people find out about this project?

**Comment by MarkJaroski** [comment]
Maybe a croncert bot in the Fediverse?

**Reply by MarkJaroski** [reply]
In fact I've been wondering about events in the Fediverse in general. Venues seem to like using things like Instagram and FB for their events, why not attempt to liberate that stuff?

**Reply by jakopako** [reply]
Are you talking about motivating venues to start using the Fediverse or, as you mention in one of the other comments, make concertcloud known via another channel?

**Reply by MarkJaroski** [reply]
A little bit of both I think. Somebody has to lure them away from Meta, which is trying to establish some kind of monopoly.

**Comment by MarkJaroski** [comment]
Meanwhile expect new Basel and Berlin venues ... ;)

**Reply by jakopako** [reply]
Awesome, thanks! I'll also have to dedicate some time to this project again. Probably gonna fix some configs that broke due to website changes.

**Comment by MarkJaroski** [comment]
@jakopako, do you have a Fediverse presence? Maybe there should be an official Croncert account in some Mastodon instance or the other. We have https://tooting.ch here in Switzerland to start with.

**Reply by MarkJaroski** [reply]
I am @spraoi@tooting.ch among other identities.

**Reply by jakopako** [reply]
Hey @MarkJaroski thanks for taking the time to comment on this issue! The Fediverse might indeed be a good channel for croncert (or concertcloud.live. I renamed it at some point because no one seemed to quite understand or could remember croncert). I'm not really a social media person though so I don't have a Fediverse presence. But if you want to have a go at creating such an account, feel free!

**Comment by MarkJaroski** [comment]
Is there some way to make a URL for a particular search setting?

**Reply by jakopako** [reply]
Unfortunately not yet.. but there is an issue 😅 https://github.com/jakopako/croncert-web/issues/38

**Reply by jakopako** [reply]
This has been fixed, btw

**Comment by MarkJaroski** [comment]
FYI: https://tooting.ch/@concertcloud

**Reply by jakopako** [reply]
Awesome initiative!

**Comment by MarkJaroski** [comment]
So, since individual events in the API do not (yet) have a unique ID I think I'll do something like this:

event JSON | cannonical representation | hash

And then store the hashes somewhere, say in a hashes.txt file (to allow this to be deployed as a GitHub action). 

Then, to avoid reposting an event I can just grep the hashes.txt file. This will have the useful side effect of reposting an event which has changed.

**Reply by jakopako** [reply]
Sounds good. At some point I'll definitely improve the api and how events are being written to the api. Currently, the mechanism is rather stupid im order to make sure that there won't be duplicates in case the scraped data is not 'clean' ( eg the title of an event suddenly changes slightly bit it's still the same event). Basically, what happens is that all the future events in the database are deleted before they're being inserted again. So the internal DB id for an event changes every time the scraper runs. I'll definitely improve this at some point. 

Anyways, looking forward to your bot!

**Comment by MarkJaroski** [comment]
Here's a first test of what these things are likely to look like:

https://tooting.ch/@concertcloud/111291151031712649

and 

https://tooting.ch/@concertcloud/111291150692001742

**Comment by MarkJaroski** [comment]
@jakopako have you seen this? https://docs.joinmobilizon.org/contribute/graphql_api/

And this? https://mobilisons.ch/

**Reply by jakopako** [reply]
Interesting, haven't heard of this before. I'll have to check this out in more detail.

**Comment by MarkJaroski** [comment]
The admin of tooting.ch is encouraging me to dump events from the concertcloud.live API into Mobilisons, but I'm wondering if their API might actually be a reasonable target for the scrapers in the first place. ...

**Reply by jakopako** [reply]
Interesting thoughts! I'll make myself more familiar with mobilisons. As far as I could see from a first glance there's no concert data in the .ch instance. So I'm not sure if it would make a lot of sense as a target for the scraper.

**Reply by MarkJaroski** [reply]
Well, there will be 1208 concerts tonight or tomorrow ;)

**Reply by jakopako** [reply]
Haha nice! You're fast :D

**Reply by MarkJaroski** [reply]
Well, the admin has asked me to keep it to 100 per hour or something.

I'll commit code shortly so you can see what I'm up to. I did the Masto bot in Python, but this one is Go.

**Reply by MarkJaroski** [reply]
It might take a bit longer. GraphQL is a little more complex than an API should be, I think.

**Reply by MarkJaroski** [reply]
I was overly optimistic. Everything in Mobilizòn is strongly typed, so converting between, say our _city_ and their search location is going to be more than one evening's effort.

**Reply by jakopako** [reply]
hmm, bummer :/ I hope it's not too complicated.

**Reply by MarkJaroski** [reply]
Well, it's not so bad once you get used to the concepts. The main thing slowing me down is the strong typing in the API, which requires some care of course. I'll get there this weekend.

**Reply by MarkJaroski** [reply]
Yesterday I did a bit more work on trying to extract addresses from Mobilizòn for use in creating events, and had a revelation. They're just proxying the much simpler and much faster OpenStreetMap Nominatum REST API.

So that's easy. 

But here's the trick: in order to make the search work 100% of the time, the venue name needs to be a pretty close match with an amenity point in OpenStreetMap. I updated all of the amenities in OpenStreetMap for Lausanne last night, and that works, but I think we'll probably need to make some adjustments in the croncert-config as well for some other cities.

**Reply by jakopako** [reply]
Ok, cool! Actually, I am also using open street map in my api to map city+country to coordinates for the radius search. So the naming changes and updates to openstreetmap might come in handy for me as well if at some point I'll try to have more accurate geolocations (venue instead of city) 🥳

**Reply by MarkJaroski** [reply]
I got the basic client to work!

Here's a first test post: https://mobilisons.ch/events/95ed2ecc-bdea-4868-8ecf-ea781631f2d2

**Reply by MarkJaroski** [reply]
And over on tooting.ch (Mastodon) it looks like this:

![image](https://github.com/jakopako/croncert-config/assets/79238/44b3eb07-9371-46cc-bb25-deb22a631e33)

**Reply by jakopako** [reply]
Nice, great work! I like that you link to the contribute page.

**Reply by MarkJaroski** [reply]
I just uploaded all of today's events in Switzerland. I'm doing it one-by-one because I'm still learning how Mobilizon displays the data. Also, I have no really good way to automate the images so far.

https://mobilisons.ch/@concert_cloud_switzerland/events

It would be nice to add an image_url field maybe, to support this kind of thing.

**Reply by MarkJaroski** [reply]
A question: how frequently does the scraper run these days?

**Reply by jakopako** [reply]
It currently runs during Saturday night and Tuesday night, see https://github.com/jakopako/croncert-config/blob/main/.github/workflows/main.yml#L5

How do you currently add the images? I guess we could add a field image_url to the scraper config & event-api, but it would involve a lot of manual work to update the scraper config accordingly.

I have looked at the events on mobilison. I really like how the location is the actual venues location. Probably you've noticed this already but the times seem to be wrong. At least for https://mobilisons.ch/events/249161d2-6be9-435f-80b5-1f9ee96be7ac on mobilison it's at 10pm while in reality it's at 11pm. Probably time zone related.

**Reply by MarkJaroski** [reply]
I'll do some checks on whether this happens at the unmarshalling of the date from json or at the event creation, and see if I can make it more accurate. I eventually want to expand this to other Fediverse servers in different markets, so I have to make sure the timezone stuff is solid.

For today, I've done the images completely manually. That's obviously not sustainable at scale. I was hoping to find some kind of library where somebody had already solved the problem of finding the most appropriate image for a gig, but no luck.

**Reply by MarkJaroski** [reply]
It goes without saying that I would be motivated to update the scraper config, as I add Fediverse servers.

**Reply by MarkJaroski** [reply]
I see. Mobilisons.ch had interpreted the date correctly, but since I didn't give it a timezone it used the one from the date, which was UTC.

**Reply by MarkJaroski** [reply]
It wouldn't be a bad idea, I think, to expose the TZ data from the scraper in the API. There are some areas which span two timezones. For instance if you extend Chicagoland to North Bend and Perdue it crosses the EST/CST boundary.

That's an extreme example of course.

**Reply by jakopako** [reply]
You're right, it's a good idea to expose the TZ data. Haven't thought of this yet since my only use case was people requesting data through the browser where the timezone is the current user's timezone. I'll make an issue in https://github.com/jakopako/event-api

**Reply by MarkJaroski** [reply]
FYI: I've started loading _all_ events from Suisse Romande.

https://mobilisons.ch/@concert_cloud_switzerland/events?page=1

**Reply by jakopako** [reply]
Very cool! Did you solve the image problem?

**Comment by MarkJaroski** [comment]
I wouldn't really call it solved.

What I decided to do was to take whatever open graph URL was offered as
canonical so long as it exists, and then to take the biggest image on the
sub-page for the event.

In short, it works most of the time. I'd rather have it in the Concert
Cloud data.

But anyway, it's coming along.

**Reply by jakopako** [reply]
Nice! Btw, the imageUrl field is supported now by the event api.

**Comment by MarkJaroski** [comment]
I should mention that I'm in Neuchatel today meeting with somebody from Fairsocial.net.

We should organise a bigger meetup at some point.
